src/SFA.DAS.ApprenticePortal.Web/Pages/Home.cshtml.cs

```diff
@@ -28,18 +30,19 @@ public async Task OnGet()
             try
             {
                 if (!_notifications.Notifications.Contains(HomeNotification.ApprenticeshipDidNotMatch))
-                    CourseName = await GetCourseName();
+                    await PopulateProperties();
             }
             catch
             {
                 _notifications.Notifications.Add(HomeNotification.ApprenticeshipDidNotMatch);
             }
         }
 
-        private async Task<string> GetCourseName()
+        private async Task PopulateProperties()
         {
             var apprenticeships = await _client.GetApprenticeships(_user.ApprenticeId);
-            return apprenticeships.Apprenticeships[0].CourseName;
+            CourseName = apprenticeships.Apprenticeships[0].CourseName;
+            ApprenticeshipComplete = apprenticeships.Apprenticeships[0].ConfirmedOn.HasValue;

```

Comment:
I worry a little about this pattern, will it error if there are no apprenticeships?

Would it be better if this implemented in a service, then the intent becomes clearer and also could have an explicit test. Also it's expensive getting all apprenticeships and all revisions just to make this call and everytime we land on this page we make this call.

Would it be possible to use a REST style such as

apprentices\XXXXX\apprenticeships\latest\revisions\latest

which would only get one single record. Failing that the other option would be to cache the result and only call this periodically.

Reply:
This PR isn't changing how the API is called, is it?  It's just using more data that was already being returned.

As an enhancement, adding a `/latest/` route to the API could be a useful addition.  I think the HTTP call itself *probably* has a larger overhead than the amout of data returned - especially once it's gone through two layers of APIM - so I wouldn't prematurely optimise the result list until we do some analysis.

Maybe a focus could be on ensuring there's only *one* API call per page load, I suspect that would have a larger impact, but again - needs analysis to show where the slow parts are.

I think any of that should be part of a different ticket though, not this one.